source/extensions/filters/udp/dns_filter/dns_impl.cc

```diff
@@ -0,0 +1,311 @@
+#include "extensions/filters/udp/dns_filter/dns_impl.h"
```

Comment:
This is largely the same as implementation in source/common/network/dns_impl.cc (looks like the main differences are an addition of `DnsResolverImpl::createDnsResolver()` and removal of a main thread assert in onAresGetAddrInfoCallback). Couldn't you most of the implementation in a base class and use subclasses in common/network package and here?

Reply:
yeah, that makes sense, onAresGetAddrInfoCallback is not a virtual method to override though. Let me see if I can change it to use error propagation without exception.